### PR TITLE
Issue #675

### PR DESCRIPTION
**Pull request by boneyard93501**
added the ConciseContract namespace collision check, test function, and mangled a simple cut & paste job to update the reminder that ConciseContract leads to loss of Contract methods; move the _none_addr from ConciseContract to standalone.
there is some room for improvement in the collision test case for staging the AssertionError and i revisit that sunday or monday.

### CODE REVIEW

File: tests/core/contracts/test_concise_contract.py [hunk 1]
```diff
@@ -102,3 +102,25 @@ def test_conciscecontract_keeps_custom_normalizers_on_base(web3):
...
+
+    def getValue():
+        assert 'getValue' in STRING_ABI
+
+    setattr(ConciseContract, 'getValue', getValue)
+    concise_contract = web3.eth.contract(address=some_address,
+                                         abi=STRING_ABI,
+                                         ContractFactoryClass=ConciseContract)
+
+    with pytest.raises(AttributeError, match=r'Namespace collision .* with ConciseContract API.'):
```
Comment (pipermerriam) [inline tests/core/contracts/test_concise_contract.py]: TIL `pytest.raises` takes a `match` parameter!

File: web3/contract.py [hunk 2]
```diff
@@ -638,6 +638,14 @@ def _encode_constructor_data(cls, args=None, kwargs=None):
         return deploy_data
 
 
+def mk_collision_prop(fn_name):
+    def collision_fn():
+        msg = "Namespace collision for function name {0} with ConciseContract API.".format(fn_name)
+        raise AttributeError(msg)
+    collision_fn.__name__ = fn_name
+    return functools.partial(collision_fn)
```
Comment (pipermerriam) [inline web3/contract.py]: I'm not sure this `partial` is needed.

Comment (boneyard93501) [reply web3/contract.py]: you're right. it's a lazy left-over from when the collision_fn took signature still included  _self_ . thanks.

**Comment by pipermerriam** [comment]
Fixes https://github.com/ethereum/web3.py/issues/675

**Comment by carver** [comment]
@boneyard93501 sorry for the delay on this one! I merged in master to deal with the conflicts. Can you check that nothing got mangled in the merge? Feel free to do a rebase on master instead, and drop the merge. If I don't hear back for a few days (and CI turns green), then I'll go ahead and merge this as is.

**Comment by boneyard93501** [comment]
thx. out of pocket for a couple more days but then with full speed.

**Comment by boneyard93501** [comment]
@carver i rebased that few days ago. want me to resist contract.rts or are you good?

**Comment by dylanjw** [comment]
Im currently working on this rebase, which results in failed tests.

I think the issue is with a bug introduced in some recent changes, and to do with this PR (outside of the fact that this PR seems to be successfully triggering the bug).  I will post an issue when I have a better understanding of what is happening.

Essentially, the issue is that `contract.abi` is from a different contract than that which is supplied.

**Comment by carver** [comment]
> want me to resist contract.rts or are you good?

Sorry, what?

Looks like the only thing left is a small lint fixup:

> /home/circleci/repo/tests/core/contracts/test_concise_contract.py:25:1: E302 expected 2 blank lines, found 1

**Comment by dylanjw** [comment]
I went ahead and did a rebase yesterday. It required reconciling with some recent changes.  Ive submitted the lint fix (my mistake).

**Pull request merged**